src/test/analyserwaveformtest.cpp

```diff
@@ -63,7 +63,12 @@ class AnalyzerWaveformTest : public MixxxTest {
 
         if (actual.size() == reference.size()) {
             for (int i = 0; i < actual.size(); ++i) {
-                if (actual[i] != reference[i]) {
+                // oprerator[] returns a QByteRef holding the index and a
+                // reference to the original array. The != operator is not
+                // explicit defined and considered as ambiguous
+                // [-Werror,-Wambiguous-reversed-operator]
+                // The operator!=(char c) is defined.
+                if (actual[i] != static_cast<char>(reference[i])) {
```

Comment:
not sure I'm comfortable with the cast here to be honest. If you refactor the the declaration of `reference` to be const, `operator[]` will return a char instead of a `QByteRef`. QByteRef is a bit weird anyways IMO since there is no documentation for it. Would probably be better to just make both `QByteArrays` const. 

```cpp
        ASSERT_TRUE(f.open(QFile::ReadOnly));
        QByteArray const reference = f.readAll();
```

Reply:
Oh yes, good idea. I have also used at() to make clear that I want only the char. This way we can also remove the comment. 

Regarding the const position, we follow the Google style recommendation in Mixxx that recommends putting const first: https://google.github.io/styleguide/cppguide.html#Use_of_const

Reply:
> Oh yes, good idea. I have also used at() to make clear that I want only the char. This way we can also remove the comment.

:+1: 

> Regarding the const position, we follow the Google style recommendation in Mixxx that recommends putting const first: https://google.github.io/styleguide/cppguide.html#Use_of_const

The style guide only recommends to add const, not particularly where IMO. I think we should prefer east const in the future. https://isocpp.org/wiki/faq/const-correctness#const-ref-alt